eth/fetcher/fetcher.go

```diff
@@ -204,6 +211,10 @@ func (f *Fetcher) Notify(peer string, hash common.Hash, number uint64, time time
 
 // Enqueue tries to fill gaps the fetcher's future import queue.
 func (f *Fetcher) Enqueue(peer string, block *types.Block) error {
+	if f.hasBlock(block.Hash()) {
+		return nil
+	}
+	propBroadcastUsefulMeter.Mark(1)
```

Comment:
Let's add a similar change to Notify (instead of filtering in eth/handler.go). Also lets track total announces/broadcasts and filtered out ones similat to txs.